core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -2440,6 +2442,57 @@ class KafkaApisTest {
     assertNull(partitionData.abortedTransactions)
   }
 
+  /**
+   * Verifies that partitions with unknown topic ID errors are added to the erroneous set and there is not an attempt to fetch them.
+   */
+  @Test
+  def testFetchRequestErroneousPartitions(): Unit = {
+    val tidp = new TopicIdPartition(Uuid.randomUuid(), new TopicPartition("foo", 0))
+    val tp = tidp.topicPartition
+    val nullTp = new TopicPartition(null, tp.partition)
```

Comment:
Could we simplify all of that by defining two `TopicIdPartition`? For instance, we could have the following:

```
val foo = new TopicIdPartition(Uuid.randomUuid(), new TopicPartition("foo", 0));
val unresolvedFoo = new TopicIdPartition(foo.topicId, new TopicPartition(null, foo.partition));
```

Then, we can use them where we need them.